manifests/init.pp

```diff
@@ -94,6 +104,17 @@
   $middleware_ssl_cert_path = "${ssldir}/middleware_cert.pem"
   $middleware_ssl_ca_path   = "${ssldir}/middleware_ca.pem"
 
+  if $securityprovider == 'sshkey' {
+    package{'sshkeyauth':
+      ensure   =>  $sshkeyauth_gem_version,
+      provider =>  'puppet_gem',
```

Comment:
This needs to be conditionalized. I don't think this provider will work on Puppet <4.0. Provider for Puppet 3.x should be 'gem'.